src/main/java/seedu/partyplanet/storage/JsonAdaptedPerson.java

```diff
@@ -114,13 +114,14 @@ public Person toModelType() throws IllegalValueException {
         }
         if (birthday.equals(Birthday.EMPTY_BIRTHDAY_STRING)) {
             modelBirthday = Birthday.EMPTY_BIRTHDAY;
-        }
-        try {
-            modelBirthday = new Birthday(birthday);
-        } catch (DateTimeException err) { // birthday year exceeds current year
-            throw new IllegalValueException((Birthday.MESSAGE_YEAR_CONSTRAINTS));
-        } catch (IllegalArgumentException err) { // date in wrong format
-            throw new IllegalValueException(Birthday.MESSAGE_CONSTRAINTS);
+        } else {
+            try {
+                modelBirthday = new Birthday(birthday);
+            } catch (DateTimeException err) { // date in wrong format
+                throw new DateTimeException(Birthday.MESSAGE_CONSTRAINTS);
```

Comment:
IllegalValueException was correct haha
IllegalValueException is the one caught by storage. Or else an invalid date in the save file will not be handled and program will terminate

```suggestion
                throw new IllegalValueException(Birthday.MESSAGE_CONSTRAINTS);
```

![image](https://user-images.githubusercontent.com/71179511/111106751-a4b45a80-8590-11eb-96f8-92ddc76326dd.png)